leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/RegisterResource.java

```diff
@@ -254,9 +254,9 @@ private Identity extractIdentity(CoapExchange exchange) {
                 return Identity.rpk(peerAddress, publicKey);
             } else if (senderIdentity instanceof X500Principal) {
                 // Extract common name
-                Matcher endpointMatcher = Pattern.compile("CN=.*?,").matcher(senderIdentity.getName());
+                Matcher endpointMatcher = Pattern.compile("CN=(.*?)(,|$)").matcher(senderIdentity.getName());
                 if (endpointMatcher.find()) {
-                    String x509CommonName = endpointMatcher.group().substring(3, endpointMatcher.group().length() - 1);
+                    String x509CommonName = endpointMatcher.group(1);
```

Comment:
What do you mean with "twice"?
One for the client one for the server?
We only have "client-cf" and "server-cf" which contains the californium specific stuff.
A common package for "cf" is currently not used.